### PR TITLE
IC-1815: Remove references for old single referral fields on SP EOSR journey

### PR DESCRIPTION
**Pull request by andrewburrowsmoj**
## What does this pull request do?

Replaces use of serviceCategoryId with serviceCategoryIds, desiredOutcomeIds with desiredOutcomes and complexityLevelId with complexityLevels for SP EOSR journey.

## What is the intent behind these changes?

Provide base to use new cohort structure for SP EOSR journey

### CODE REVIEW

**Comment by andrewburrowsmoj** [comment]
I've "cohortized" the SP EOSR outcomes page. 
However, I've not done the same for FurtherInformation the EOSR form page and EOSR confirmation.

It seems that it would be easy to do and I've laid the framework for it in this PR; however I need confirmation on the correct solution.

Currently these pages use `serviceCategory.name`.

So to "cohortize" it, we could either do:
`serviceCategory.getAllNames().join(",")`
or
`intervention.contractType.name`

File: integration_tests/integration/referral.spec.js
Comment (tomsmyers) [inline integration_tests/integration/referral.spec.js:514]: small nit: this service type should be "women's services" i think

**Review by tomsmyers — COMMENTED** [review]
this all looks good to me but i'm going to defer approval to george or lawrence since i haven't touched the EOSR portion of the codebase before; it's not all that familiar to me.

File: server/routes/serviceProviderReferrals/serviceProviderReferralsController.ts
Comment (Gweaton) [inline server/routes/serviceProviderReferrals/serviceProviderReferralsController.ts:819]: I wonder if this is something that could go on a `referralDecorator` object (like the `interventionDecorator` that Lawrence introduced recently) that looks through its serviceCatgegoryIds`

**Review by Gweaton — APPROVED** [review]
This looks fine to me, just one comment:

File: server/routes/serviceProviderReferrals/serviceProviderReferralsController.ts
Comment (andrewburrowsmoj) [reply server/routes/serviceProviderReferrals/serviceProviderReferralsController.ts:819]: yes, I left a comment on Lawrence's PR with exactly that suggestion! However, I think the class would need to be renamed.

Comment (andrewburrowsmoj) [reply server/routes/serviceProviderReferrals/serviceProviderReferralsController.ts:819]: However - he hasn't been able to merge in his PR just yet. Would it be okay for me to make that change on a future PR?

Comment (lawrence-forooghian) [reply server/routes/serviceProviderReferrals/serviceProviderReferralsController.ts:819]: I've merged mine now. I've actually already written a method that does the same thing as this, on the branch that I'm working on. But let's not block this PR - once I've got that method merged then I'll do a scan through the codebase for things that can be updated.

Comment (Gweaton) [reply server/routes/serviceProviderReferrals/serviceProviderReferralsController.ts:819]: Yep, let's get this in

**Review by lawrence-forooghian — APPROVED** [review]

**Pull request merged**
